File: chess_frontend/src/app/models/game.ts
export class Move{
    moveNumber: number;
    player: string;
    notation: string;

    constructor(moveNumber: number, player: string, notation: string){
        this.moveNumber = moveNumber;
        this.player = player;
        this.notation = notation;
    }
}

export class Game {
    gameCode: string;
    playerWhiteId: number;
    playerBlackId: number;
    status: string;
    created: Date;
    completed: Date;
    winnerId: number;
    moveHistory: Move[];

    constructor(gameCode: string, playerWhiteId: number, playerBlackId: number, status: string, created: Date, completed: Date, winnerId: number, moveHistory: Move[]){
        this.gameCode = gameCode;
        this.playerWhiteId = playerWhiteId;
        this.playerBlackId = playerBlackId;
        this.status = status;
        this.created = created;
        this.completed = completed;
        this.winnerId = winnerId;
        this.moveHistory = moveHistory;
    }
}   

File: chess_backend/chess_backend/EFCore/Game.cs
namespace chess_backend.EFCore
{
    public class Move
    {
        public int Id { get; set; }
        public int GameId { get; set; }      // FK
        public Game Game { get; set; }      // back-reference
        public int MoveNumber { get; set; }
        public int Player { get; set; }
        public string Notation { get; set; }
    }

    public class Game
    {
        public int Id { get; set; }
        public string GameCode { get; set; }
        public int PlayerWhiteId { get; set; }
        public int PlayerBlackId { get; set; }
        public string Status { get; set; }
        public DateTime Created { get; set; }
        public DateTime Completed { get; set; }
        public int WinnerId { get; set; }
        public ICollection<Move> MoveHistory { get; set; } = new List<Move>();
    }
}
